src/ralph/data_center/tests/test_subscribers.py

```diff
@@ -0,0 +1,217 @@
+
+        self.data['port'] = None
+        errors = validate_vip_event_data(self.data)
+        self.assertEqual(len(errors), 1)
+        self.assertIn('invalid port "None"', errors)
+
+    def test_missing_protocol(self):
+        self.data['protocol'] = None
+        errors = validate_vip_event_data(self.data)
+        self.assertEqual(len(errors), 1)
+        self.assertIn('missing protocol', errors)
+
+    def test_missing_service_uid(self):
+        self.data['service']['uid'] = None
+        errors = validate_vip_event_data(self.data)
+        self.assertEqual(len(errors), 1)
+        self.assertIn('missing service UID', errors)
+
+        self.data['service'] = None
+        errors = validate_vip_event_data(self.data)
+        self.assertEqual(len(errors), 1)
+        self.assertIn('missing service UID', errors)
+
+    def test_missing_environment(self):
+        self.data['environment'] = None
+        errors = validate_vip_event_data(self.data)
+        self.assertEqual(len(errors), 1)
+        self.assertIn('missing environment', errors)
+
+    def test_if_errors_aggregate(self):
+        self.data['service'] = None
+        self.data['environment'] = None
+        errors = validate_vip_event_data(self.data)
+        self.assertEqual(len(errors), 2)
+        self.assertIn('missing service UID', errors)
+        self.assertIn('missing environment', errors)
+
+
+class HandleCreateVIPEventTestCase(TestCase):
+
+    def setUp(self):
+        logging.disable(logging.CRITICAL)
+        self.data = deepcopy(EVENT_DATA)
+
+    def test_create_when_vip_already_exists(self):
+        vip = VIPFactory()
+        self.data['ip'] = vip.ip.address
+        self.data['port'] = vip.port
+        self.data['protocol'] = VIPProtocol.from_id(vip.protocol).name
+
+        vips = VIP.objects.all()
+        self.assertEqual(vips.count(), 1)
+        modified_before = vips[0].modified
+        handle_create_vip_event(self.data)
+        vips = VIP.objects.all()
+        self.assertEqual(vips.count(), 1)
+        self.assertEqual(vips[0].modified, modified_before)
+
+    def test_create_when_service_env_does_not_exist(self):
+        self.data['service']['uid'] = 'non-existing-uid'
+        handle_create_vip_event(self.data)
+        self.assertEqual(VIP.objects.count(), 0)
+
+    def test_create_with_valid_event_data(self):
+        service_env = ServiceEnvironmentFactory()
+        self.data['service']['uid'] = service_env.service.uid
+        self.data['environment'] = service_env.environment.name
+        self.assertEqual(VIP.objects.count(), 0)
+        handle_create_vip_event(self.data)
+        self.assertEqual(VIP.objects.count(), 1)
+
+    def test_create_with_invalid_event_data(self):
+        self.data['service'] = None
+        self.assertEqual(VIP.objects.count(), 0)
+        handle_create_vip_event(self.data)
+        self.assertEqual(VIP.objects.count(), 0)
+
+    def tearDown(self):
+        logging.disable(logging.NOTSET)
```

Comment:
So what about disabling all loggers during tests, on all levels? Are you OK with that..?